app/src/test/kotlin/fr/free/nrw/commons/category/CategoriesModelTest.kt

```diff
@@ -17,6 +20,18 @@ class CategoriesModelTest {
     @Mock
     internal var categoryItem: CategoryItem? = null
 
+    @Spy
+    internal var gson: Gson? = Gson()
+
+    @Spy
+    internal var categoryItemForMethodA: CategoryItem? = CategoryItem("",false)
+
+    @Mock
+    internal var categoryDao: CategoryDao? = null
```

Comment:
Could use lateinit for mocks

Reply:
did that wherever possible but rest of the dependencies are quite interdependent and complex